src/pocketmine/level/Level.php

```diff
@@ -2713,6 +2772,107 @@ public function setSeed(int $seed){
 		$this->provider->setSeed($seed);
 	}
 
+	/**
+	 * Gives the current weather type/id
+	 *
+	 * @return int
+	 */
+	public function getWeather() : int{
+		return $this->weather;
+	}
+
+	/**
+	 * Gives the time until the next rain in ticks
+	 *
+	 * @return int
+	 */
+	public function getRainTime() : int{
+		return $this->rainTime;
+	}
+
+	/**
+	 * Gives the time until the next thunderstorm in ticks
+	 *
+	 * @return int
+	 */
+	public function getThunderTime() : int{
+		return $this->thunderTime;
+	}
+
+	/**
+	 * Gives the time until the next clear sky in ticks
+	 *
+	 * @return int
+	 */
+	public function getClearTime() {
+		return $this->clearTime;
+	}
+
+	/**
+	 * Sets the type of weather being displayed
+	 *
+	 * @param int $weatherType
+	 */
+	public function setWeather(int $weatherType){
+		$this->getServer()->getPluginManager()->callEvent($ev = new WeatherChangeEvent($this, $this->weather, $weatherType));
+		
+		if($ev->isCancelled()){
+			return;
+		}
+
+		$this->weather = $ev->getNewWeather();
+		if($ev->getNewWeather() >= self::WEATHER_RAIN){
+			$this->rainTime = 0;
+			if($ev->getNewWeather() == self::WEATHER_RAIN_THUNDER){
+				$this->thunderTime = 0;
+			}
+			$this->clearTime = mt_rand((self::TIME_FULL / 2), (7 * self::TIME_FULL) + (self::TIME_FULL / 2));
+		}elseif($ev->getNewWeather() == self::WEATHER_NORM){
+			$this->rainTime = mt_rand((self::TIME_FULL / 2), (7 * self::TIME_FULL) + (self::TIME_FULL / 2));
+			$this->thunderTime = mt_rand((self::TIME_FULL / 2), (7 * self::TIME_FULL) + (self::TIME_FULL / 2));
+			$this->clearTime = 0;
+		}
+		
+		foreach($this->getPlayers() as $player){
+			$player->sendWeather();
+		}
+	}
+
+	/**
+	 * Sets the time until the next rain in ticks
+	 *
+	 * @param int $time
+	 */
+	public function setRainTime(int $time) {
+		$this->rainTime = $time;
+	}
+
+	/**
+	 * Sets the time until the next thunderstorm in ticks
+	 *
+	 * @param int $time
+	 */
+	public function setThunderTime(int $time) {
+		$this->thunderTime = $time;
+	}
+
+	/**
+	 * Sets the time until the next clear sky in ticks
+	 *
+	 * @param int $time
+	 */
+	public function setClearTime(int $time) {
+		$this->clearTime = $time;
+	}
+
+	/**
+     * Sets wether or not Weather is updated each tick
```

Comment:
And possibly would be better to be `Sets whether or not the weather is updated each tick`. Notice the added `the`